src/main/config_refresher.rs

```diff
@@ -78,23 +78,22 @@ impl<UID: Uid> State for ConfigRefresher<UID> {
                 }
             };
 
-        let config = match read_config_file() {
-            Ok(cfg) => cfg,
+        match self.config.reload_and_check_modified() {
+            Ok(true) => (),
+            Ok(false) => return,
```

Comment:
I might be misunderstanding code, but i think the overall logic is now incorrect. So if there was an update to config file and then someone bootstrapped off us, in connection-listener/exchange-msg, we would have reloaded the config file from the disk. Assume the new config has added the bootstrapper to the whitelist and has removed some other peer IP from the whitelist. Also say this removed peer was already an ActiveConnection. Now when the above condition is evaluated it's going to return `Ok(false)` because the in-mem and on-disc configs are the same. So the ActiveConnection which is no longer in whitelist will not be removed and if no other events occur, it will continue to remain so indefinitely. This can be an attack-vector.

In the existing code in master all these are taken care of and code would continue below to eject the ActiveConnection.